### ISSUE TITLE
Failed to change partition state (arming modes bug)

### ISSUE THREAD

**Issue by danemacmillan** [node-alarm-dot-com/homebridge-node-alarm-dot-com#16]
Arming and disarming the panel is not working as it used to. These are the errors in my log:

![223AB3A5-13B9-42FD-9B3F-A2EEC38F851E](https://user-images.githubusercontent.com/4390491/60775939-22567080-a0f6-11e9-88a6-3e8e405e5c00.jpeg)
![1D031146-4063-4790-AC55-CE6679BBE6E9](https://user-images.githubusercontent.com/4390491/60775940-22567080-a0f6-11e9-96bd-de167f566785.png)

**Comment by mkormendy** [comment]
This happens every time you reboot?

**Comment by danemacmillan** [comment]
This happens any time I attempt to use the accessory in HomeKit.

**Comment by mkormendy** [comment]
How about the other answers.

**Comment by danemacmillan** [comment]
Hey--at the time it was the only one in the email.

> This happens every time you restart homebridge?

Restarting doesn't change the result. 

> What relevant hardware, software versions are you running this on?

`Linux localhost 3.0.31-1769538 #1 SMP PREEMPT Mon Sep 30 22:19:10 KST 2013 armv7l armv7l armv7l GNU/Linux`

It's an Ubuntu Trusty version, but I'll have to edit to include the version later. It's running on an old Android tablet that is running a VM.

I've restarted it a few times, and I've also deleted the accessory cache, but to no avail.


Here's the tail end of output immediately after restarting, then attempting to arm the system for stay.


```
[2019-7-8 07:41:35] [Security System] Logging into Alarm.com as alarmforcehomebridge@danemacmillan.com
[2019-7-8 07:41:35] [CMD Switch] Initializing platform accessory 'Stream Fix'...
[2019-7-8 07:41:35] Homebridge is running on port 51826.
[2019-7-8 07:41:38] [Security System] Logged into Alarm.com as alarmforcehomebridge@danemacmillan.com
[2019-7-8 07:41:39] [Security System] Received 1 partitions(s) and 7 sensor(s) from Alarm.com
[2019-7-8 07:41:39] [Security System] Removing Panel (94902417-127) from HomeBridge.
[2019-7-8 07:41:39] [Security System] Adding partition Panel (id=94902417-127, uuid=ad21369c-5836-4683-83be-15ab565d536c)
[2019-7-8 07:41:39] [Security System] Updating partition Panel (94902417-127), state=3, prev=null
[2019-7-8 07:41:39] [Security System] Updating partition Panel (94902417-127), desiredState=3, prev=null
[2019-7-8 07:41:39] [Security System] Updating partition Panel (94902417-127), statusFault=false, prev=null
[2019-7-8 07:41:39] [Security System] Added partition Panel (94902417-127)
[2019-7-8 07:41:39] [Security System] Ignored sensor Basement Hallway Motion Detector (94902417-6)
[2019-7-8 07:41:39] [Security System] Ignored sensor Bedroom Motion Detector (94902417-4)
[2019-7-8 07:41:39] [Security System] Ignored sensor Front Door (94902417-1)
[2019-7-8 07:41:39] [Security System] Ignored sensor Left Window Bedroom (94902417-2)
[2019-7-8 07:41:39] [Security System] Ignored sensor Right Bedroom Window (94902417-3)
[2019-7-8 07:41:39] [Security System] Ignored sensor Upstairs Motion Detector (94902417-5)
[2019-7-8 07:41:39] [Security System] Ignored sensor Upstairs Smoke Detector (94902417-7)
[2019-7-8 07:43:40] [CMD Switch] Stream Fix is off.
[2019-7-8 07:44:00] [Security System] changePartitionState(94902417-127, 0)
[2019-7-8 07:44:00] [Security System] Error: Failed to change partition state: Error: POST https://www.alarm.com/web/api/devices/partitions/94902417-127/armStay failed: 422
    at fetch.then.then.catch.err (/usr/lib/node_modules/homebridge-node-alarm-dot-com/node_modules/node-alarm-dot-com/index.js:360:13)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7)
[2019-7-8 07:44:00] [Security System] changePartitionState(94902417-127, 3)
```

**Comment by mkormendy** [comment]
What version of node are you running?

**Comment by danemacmillan** [comment]
I'm running `v8.12.0`.

**Comment by mkormendy** [comment]
Hrmm .. can you try and upgrade your node to 12.4.0?  I've been on >10 since I started this project.

**Comment by danemacmillan** [comment]
Sure, I'll see what happens tonight and let you know.

**Comment by danemacmillan** [comment]
The highest version I was able to upgrade to was `v9.3.0`. The issue persists. For reference, I've been running this module with `v8.12.0` for close to a year. This machine I run this module on doesn't get updated automatically, so it's pretty static.

I'll give it a bit more time this weekend to see if I can compile the dependencies that would get me on `12`.

**Comment by Drumroll** [comment]
Did anything happen to your network configuration recently (ie firmware updates)? Did you check your firewall settings?

**Comment by davidvanvo** [comment]
Did anything happen to your network configuration recently (ie firmware updates)? Did you check your firewall settings?

**Comment by mkormendy** [comment]
This also could be related to the plan type you have with your provider. Did that change or did they update something or change their service offering on you?

**Comment by danemacmillan** [comment]
I finally had another moment to look into this, and figured it out. I went into the file and `console.log`ged everything, and this stood out to me:

![](https://i.imgur.com/T0D075a.png)

```
errors:
   [ { status: '422',
       detail: '"Invalid arming options for the current arming state"',
       code: 422,
       source: [Object] } ],
```

The "invalid arming options" lead me to review my config, and I noticed that I had set the `silentArming` arming mode options to `true` for each mode, being `away`, `night`, and `stay`. I instead changed them all to false to mirror what the example config shows.

Apparently, setting those to true screws with the request. I don't think the options ever worked, but the changes to `true` for them clearly slipped through the cracks while I was making general changes to the config, and for whatever reason, I neglected to remember making this change or correlate it with the sudden non-functioning of the alarm system through this plugin.

For reference, I originally subscribed to Alarm Force, which was then acquired and rebranded by Bell in Canada.
